mlrun/utils/azure_vault.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2021 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from os.path import expanduser
+
+from ..config import config as mlconf
+from .helpers import logger
+
+
+class AzureVaultStore:
+    def __init__(self, vault_name):
+
+        try:
+            from azure.identity import EnvironmentCredential
+            from azure.keyvault.secrets import SecretClient
+        except ImportError as exc:
+            raise ImportError(
+                f"Azure key-vault libraries not installed, run pip install azure-identity azure-keyvault-secrets, {exc}"
+            )
+
+        self._vault_name = vault_name
+        self._url = mlconf.secret_stores.azure_vault.url.format(name=vault_name)
+        self._client = None
+
+        tenant_id = self._get_secret_file_contents("tenant_id")
+        client_id = self._get_secret_file_contents("client_id")
+
+        if not tenant_id or not client_id:
+            logger.info(
+                "both tenant_id and client_id must be configured to use Azure vault"
+            )
+            return
+
+        client_secret = self._get_secret_file_contents("secret")
+        if client_secret:
+            os.environ["AZURE_CLIENT_SECRET"] = client_secret
+
+        if "AZURE_CLIENT_SECRET" not in os.environ:
+            logger.info("Azure client secret could not be found")
+            return
+
+        # Azure EnvironmentCredential uses these environment variables. Populate them
+        os.environ["AZURE_TENANT_ID"] = tenant_id
+        os.environ["AZURE_CLIENT_ID"] = client_id
+
+        credential = EnvironmentCredential()
```

Comment:
There is no way to do it with args or something ? has to go through environment ?

Reply:
That's how Azure credentials work - there are additional options, but most of them are VS-related or more complicated. Going through environment is the easiest (and TBH, probably the only available) way to do it for our purposes. If you look here - https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/identity/Azure.Identity#defaultazurecredential, we can't use managed identity since we're not in Azure (at least not always) and not necessarily using Azure CLI or VS/VSCode for development. So, using env is the only bullet-proof approach (and it's definitely the only one for a runtime pod)